pkg/evs/controllerserver.go

```diff
@@ -266,18 +269,181 @@ func (cs *ControllerServer) ListVolumes(_ context.Context, req *csi.ListVolumesR
 
 func (cs *ControllerServer) CreateSnapshot(_ context.Context, req *csi.CreateSnapshotRequest) (
 	*csi.CreateSnapshotResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	klog.Infof("CreateSnapshot called with request %v", *req)
+	credentials := cs.Driver.cloudCredentials
+	name := req.GetName()
+	volumeId := req.GetSourceVolumeId()
+	if err := checkCreateSnapshotParamEmpty(name, volumeId); err != nil {
+		klog.Errorf("Failed to check create snapshot param, %v", err)
+		return nil, err
+	}
+	response, err := checkDuplicateSnapshotName(credentials, name, volumeId)
```

Comment:
suggest renaming `checkSnapshotNameExists`, preferably with some modifications and move to service path